ARCHITECTURE.md

```diff
@@ -36,20 +36,23 @@ This isn't the only architecture document. You may also want to look at:
 
 ## High-level overview
 
-At a high level, this repository provides two components:
+At a high level, this repository provides three components:
```

Comment:
```suggestion
At a high level, this repository provides 5 components with non-trivial amount of code:
```
1. neonvm-runner
2. neonvm-controller
3. autoscale-scheduler (plugin)
4. autoscaler-agent (side node: those autoscale vs autoscaler vs autoscaling produce mildly annoying typos)
5. vm-builder

I think it makes navigating the repo easier.

Reply:
I've updated it now. LMK what you think!

Reply:
Good!